### PR TITLE
Add `Cdc` action for Change Data Feed

### PR DESCRIPTION
**Pull request by OussamaSaoudi-db**
## What changes are proposed in this pull request?

This PR introduces the `Cdc` action and its associated `CdcVisitor`. This action is used for the Change Data Feed table feature.

## How was this change tested?

- Ensure schema projection to `cdc` works
- Ensure that the visitor correctly extracts cdc actions.
- Ensure that metadata parses enableChangeDataFeed table feature.

### CODE REVIEW

**Comment by zachschuermann** [comment]
flagging this will be impacted by #481

File: kernel/src/actions/visitors.rs [hunk 1]
```diff
@@ -243,6 +244,48 @@ impl DataVisitor for RemoveVisitor {
...
+        let size: i64 = getters[2].get(row_index, "cdc.size")?;
+        let data_change: bool = getters[3].get(row_index, "cdc.dataChange")?;
+        let tags: Option<HashMap<String, String>> = getters[4].get_opt(row_index, "cdc.tags")?;
+
+        Ok(Cdc {
+            path,
+            partition_values,
+            size,
+            data_change,
+            tags,
```
Comment (scovich) [inline kernel/src/actions/visitors.rs:272]: nit: why not just
```suggestion
        Ok(Cdc {
            path,
            partition_values: getters[1].get(row_index, "cdc.partitionValues")?,
            size: getters[2].get(row_index, "cdc.size")?,
            data_change: getters[3].get(row_index, "cdc.dataChange")?,
            tags: getters[4].get_opt(row_index, "cdc.tags")?,
```

**Review by scovich — APPROVED** [review]
Code looks fine, but what is the eventual user surface? 
`CdcVisitor` is not even `pub(crate)`?

**Comment by OussamaSaoudi-db** [comment]
@scovich good point, I'll make it mimic the visibility of other visitors/actions.

**Comment by OussamaSaoudi-db** [comment]
@scovich `CdcVisitor` has the following visibility attributes, which are the same as the `AddVisitor` and `RemoveVisitor`
```
#[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
#[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
```

**Review by nicklan — APPROVED** [review]

File: kernel/src/table_changes/table_changes_scan.rs
Comment (zachschuermann) [inline kernel/src/table_changes/table_changes_scan.rs:1]: accidentally committed?

**Review by zachschuermann — COMMENTED** [review]
a post-merge review but LGTM i think you might have accidentally merged an empty file though

**Pull request merged**